### ISSUE TITLE
Kodi official repository

### ISSUE THREAD

**Issue by nancpasc** [rolapp/plugin.video.zattooboxExt.beta#1]
Hi. I did write the ZattooBox Addon initially to fulfill my own needs, and I didn't add anything to it since it's 1st release. But it seems that you added many functionalities to it since. It would make sense to update the Addon in the Official Repository with your codebase. 

Let me know if you'd like to process further.

Best regards.
P.

**Comment by rolapp** [comment]
Hi Pascal,
This ist a good idea. How shall we call it. 'ZattooBox' or 'ZattooBoxExt'? 
Sorry for my bad english, I'm german.

Best regards
Steffen

**Comment by nancpasc** [comment]
I think that the name can't be changed, so it can be called ZattooBox still. However I'd need to update the credits, I think many people did work on this.

**Comment by rolapp** [comment]
Yes many people did work on this. How do we want to do this with the update?

**Comment by nancpasc** [comment]
I would do something like this :
1) Be sure the license files/changelogs are up to date, (in case of something changed)
2) Update the addon.xml file in order to match the original id's 
 id="plugin.video.zattoobox", version="1.X.X", name="Zattoo Box", provider-name="Steffen Rolapp,L0RE,Daniel Griner,Pascal Nançoz">
3) sent me a link to the git (stable version required). Then I gonna submit it to Kodi for a repository update.

Regards!
P.

**Comment by rolapp** [comment]
Hi, I'm finish with the work. here ist the Link
[](https://github.com/rolapp/ZattooBox/tree/stable)

**Comment by rolapp** [comment]
hi, I have made a stable repository
[ZattooBox](https://github.com/rolapp/ZattooBox)
A wiki is also ready

**Comment by nancpasc** [comment]
Hi,

Thanks, I'll have a look at it asap. I've noticed that the new License is now GPL, which is not compatible with the original BSD License. This could cause problems.,I'll have to see how to work around this.

Regards,
P.

**Comment by rolapp** [comment]
Hi,

The License from your origin is not the origin BSD License is BSD 2 - License (FreeBSD). Free BSD Licence is compatible with GNU _GPL.
[GPL-Compatible Free Software Licenses](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)

>  FreeBSD license (#FreeBSD)

    This is the original BSD license with the advertising clause and another clause removed. (It is also sometimes called the “2-clause BSD license”.) It is a lax, permissive non-copyleft free software license, compatible with the GNU GPL.

    Our comments about the Modified BSD license apply to this license too.

Regards,
Steffen

**Comment by nancpasc** [comment]
IMHO, relicensing the full project from BSD 2-clause to GPL works against the open source development ethics. http://softwareengineering.stackexchange.com/questions/75436/relicense-bsd-2-3-clause-code-to-gpl

As a reminder, here is what the first clause states :

1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer. 

I don't mind if it's relicensed to GPL, but the original license clauses must at least remain in the new license file.

**Comment by rolapp** [comment]
I have modify every code file like this.
`#    copyright (C) 2017 Steffen Rolapp (github@rolapp.de)
#
#    based on ZattooBox extended by Daniel Griner (griner.ch@gmail.com) License under GPL
#    based on ZattooBox by Pascal Nançoz (nancpasc@gmail.com) Licence under BSD 2 clause
#
#    This file is part of ZattooBox
#
#    ZattooBox is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    ZattooBox is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with ZattooBox.  If not, see <http://www.gnu.org/licenses/>.
#
`
There are two files:
license.txt ist the BSD Liscense
COPYING is a Copy of the GPL

Ich hoffe das ist Lizenz konform.

Regards Steffen

**Comment by simonneutert** [comment]
how did this work out guys?

**Comment by nancpasc** [comment]
Hi, I've been pretty busy lately. I think it would be great to be able to add this neat extension in the official Kodi repository. However, if the original addon get replaced it would be more difficult to maintain it. Furthemore I plan to fork the addon to make it compatible with other zattoo-based providers.

Maybe a better way to go is to make this extension live standalone in the repository (script, application, plugin ?) or whatever the best is, and add a dependency to the zattoobox extension. 

Not sure if possible but that would be very neat.

Best wishes.

**Comment by nancpasc** [comment]
Not sure if you fixed it yet but they did change the streaming URL. you could use the following constant in zapisession.py 

ZAPI_URL = 'https://zattoo.com'
